src/main/java/recipeapplication/controllers/RecipeController.java

```diff
@@ -58,6 +58,22 @@ public Recipe recipeDelete(@PathVariable("id") String id) {
         return this.recipes.deleteRecipe(id);
     }
 
+    @GetMapping(
+            path = "/recipeSearch/{category}",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public Recipe[] recipeSearch(@RequestParam(name = "category", required = false ) String category) {
+        return this.recipes.search(category);
+    }
+
+    @GetMapping(
+            path = "/recipeSearch/{ingredients}",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public Recipe[] recipeSearch(@RequestParam(name = "ingredients", required = false) String[] ingredients) {
+        return this.recipes.search(ingredients);
+    }
```

Comment:
this should be a single function with the path "/recipeSearch"
the signature of the method should be:
`public Recipe[] recipeSearch(@RequestParam(name = "byCategory", required = false) String category, @RequestParam(name = "byIngredients", required = false) String [] ingredients)`

Reply:
Done.